builder/gen-dockerfile/contents/src/fsview.ts

```diff
@@ -0,0 +1,56 @@
+
+import * as fs from 'fs';
+import * as path from 'path';
+
+export interface Reader {
+  read(path: string): Promise<string>
+}
+
+export interface Writer {
+  write(path: string, contents: string): Promise<any>
+}
+
+export interface Locator {
+  exists(path: string): Promise<boolean>
+}
+
+export class FsView implements Reader, Writer, Locator {
```

Comment:
Maybe http://bluebirdjs.com/docs/api/promise.promisifyall.html would be helpful here?